subsys/testsuite/ztest/src/ztest_new.c

```diff
@@ -356,13 +388,15 @@ static int run_test(struct ztest_suite_node *suite, struct ztest_unit_test *test
 	}
 	run_test_functions(suite, test, data);
 out:
-	ret |= cleanup_test(test);
-	if (test_result != ZTEST_RESULT_SUITE_FAIL) {
+	phase = TEST_PHASE_AFTER;
+	if (test_result != ZTEST_RESULT_SUITE_FAIL && !executed_after_fn) {
+		executed_after_fn = true;
 		if (suite->after != NULL) {
 			suite->after(data);
 		}
 		run_test_rules(/*is_before=*/false, test, data);
```

Comment:
Do we still want to run `rules` if anything in `after` asserts?

Reply:
I don't think so, an assert in `after` means the binary will be marked as failed. So you won't get any use from running any more test suites until you fix that issue.